test/service.cc

```diff
@@ -641,3 +641,34 @@ TEST_CASE("gVCF genotyper") {
     }
 }
 
+TEST_CASE("genotype residuals") {
+    unique_ptr<VCFData> data;
+    Status s = VCFData::Open({"discover_alleles_trio1.vcf", "discover_alleles_trio2.vcf"}, data);
+    REQUIRE(s.ok());
+    unique_ptr<Service> svc;
+    s = Service::Start(*data, *data, svc);
+    REQUIRE(s.ok());
+
+    discovered_alleles als;
+    s = svc->discover_alleles("<ALL>", range(0, 0, 1000000), als);
+    REQUIRE(s.ok());
+
+    vector<unified_site> sites;
+    s = unified_sites(als, sites);
+    REQUIRE(s.ok());
+
+    const string tfn("/tmp/GLnexus_unit_tests.bcf");
+    consolidated_loss losses;
+    genotyper_config cfg;
+    cfg.output_residuals = true;
+    s = svc->genotype_sites(cfg, string("<ALL>"), sites, tfn, losses);
+    REQUIRE(s.ok());
+
+    // verify that the residuals file is in correct yaml format
+    YAML::Node resFile = YAML::LoadFile(cfg.residuals_file);
+    REQUIRE(resFile.size() == 3);
+
+    // It seems that a list of documents split by --- symbols
+    // are parsed as a yaml map.
+    REQUIRE(resFile.IsMap());
```

Comment:
Interesting...according to [Wikipedia](https://en.wikipedia.org/wiki/YAML):

> Multiple documents can exist in a single file/stream and are separated by "---". An optional "..." can be used at the end of a file (useful for signaling an end in streamed communications without closing the pipe).

So maybe indeed it's not a top-level sequence, but there should be some way to retrieve the three different "documents." The EOF marker is a nice idea too, let's add that.